cmd/hub/mcscontroller/main.go

```diff
@@ -73,15 +78,15 @@ func main() {
 	flag.Parse()
 	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
 
-	if namespace == "" {
+	if namespace == corev1.NamespaceAll {
 		ns, err := getWatchedNamespace()
 		if err != nil {
 			setupLog.Error(err, "Failed to get watched namespace from environment")
 		}
 		namespace = ns
 	}
 
-	if namespace == "" {
+	if namespace == corev1.NamespaceAll {
 		setupLog.Error(errors.New("no namespace provided"), "Can't run in cluster scope")
```

Comment:
the handling of the namespace env / cli arg looks a bit messy to me. Had a difficulty understanding what's going on..

Reply:
cli has priority over env.
What was confusing? The use of env in the default of the flag?